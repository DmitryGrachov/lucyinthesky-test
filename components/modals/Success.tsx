import { ISuccessModal } from '../../interfaces/modals.interface';
import styles from '../../styles/success-modal.module.scss';

const Success = ({
    showModal,
    setShowModal,
    orderId,
}: ISuccessModal): JSX.Element => (
    <>
        {showModal && (
            <div className={styles.mainWrapper}>
                <div className={styles.modalWrapper}>
                    <h2>Order number</h2>
                    <p>{orderId}</p>
                    <span
                        className={styles.closebtn}
                        onClick={() => setShowModal(false)}
                    >
                        &times;
                    </span>
                </div>
            </div>
        )}
    </>
);

export default Success;
