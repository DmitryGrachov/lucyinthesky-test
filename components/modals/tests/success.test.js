import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Success from '../Success';

describe('Success component', () => {
    let expectedProps;

    beforeEach(() => {
        expectedProps = {
            showModal: true,
            orderId: 18576,
        };
    });

    test('should render orderId', () => {
        const { getByText } = render(<Success {...expectedProps} />);
        const orederId = getByText(expectedProps.orderId);

        expect(orederId).toBeVisible();
    });

    test('matches success snapshot', () => {
        const { asFragment } = render(<Success {...expectedProps} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
