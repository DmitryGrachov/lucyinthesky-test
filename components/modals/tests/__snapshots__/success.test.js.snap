// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success component matches success snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Order number
      </h2>
      <p>
        18576
      </p>
      <span>
        Ã—
      </span>
    </div>
  </div>
</DocumentFragment>
`;
